cmd/cli/id/id.go

```diff
@@ -19,27 +20,36 @@ type IDInfo struct {
 }
 
 func NewCmd() *cobra.Command {
-	OS := serve.NewServeOptions()
 	outputOpts := output.OutputOptions{
 		Format: output.JSONFormat,
 	}
 
-	// make sure serve options point to local mode
-	OS.PeerConnect = serve.DefaultPeerConnect
-	OS.PrivateInternalIPFS = true
+	idFlags := map[string][]configflags.Definition{
+		"libp2p": configflags.Libp2pFlags,
+	}
 
 	idCmd := &cobra.Command{
 		Use:   "id",
 		Short: "Show bacalhau node id info",
+		PreRun: func(cmd *cobra.Command, _ []string) {
+			if err := configflags.BindFlags(cmd, idFlags); err != nil {
+				util.Fatal(cmd, err, 1)
+			}
+		},
 		Run: func(cmd *cobra.Command, _ []string) {
-			if err := id(cmd, OS, outputOpts); err != nil {
+			if err := id(cmd, outputOpts); err != nil {
 				util.Fatal(cmd, err, 1)
 			}
 		},
 	}
 
-	idCmd.Flags().AddFlagSet(flags.OutputFormatFlags(&outputOpts))
-	serve.SetupLibp2pCLIFlags(idCmd, OS)
+	// TODO(forrest): [ux] these are flags without a corresponding value in the config
+	// in the future we can bind all flags to a config value.
+	idCmd.Flags().AddFlagSet(cliflags.OutputFormatFlags(&outputOpts))
```

Comment:
At some point in the future we may want all flags to have a corresponding entry in the config, for now I have left these out of the config (thus the package naming `cliflags` vs `configflags`) meaning the only way we can configure output, in this example, is via its flags.

Reply:
Currently (in main) we cannot configure output from env vars as far as I can tell. I have left this behavior unchanged in this PR.